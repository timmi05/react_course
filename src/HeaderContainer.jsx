import React from 'react';
import { SearchPanel } from './search/SearchPanel';
import {Product } from './product/Product';
import { AppLabel } from './component/AppLabel';
import { SearchLink } from './component/SearchLink';
import './HeaderContainer.css';

export const HeaderContainer = props => {
    const { productSelected } = props;
    const Content = productSelected ? Product : SearchPanel;

    return (
        <div className='header-container'>
            <div className='header-container-label'>
                <span>
                < AppLabel />
                </span>
                <span className='header-container-search'>
                    < SearchLink
                    productSelected = { productSelected }
                    />
                </span>
            </div>
            < Content
                runtime = '145'
                releaseDate = '2019'
                description = 'Best film of year'
                title = 'Guardians of the galaxy: VOL. 3'
                tagline = 'Tagline'
                voteAverage = '4'
                posterPath = 'https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg'
                overview = "The third film based on Marvel's Guardians of the Galaxy.
                 The third film based on Marvel's Guardians of the Galaxy. The third film based on Marvel's Guardians of
                  the Galaxy. The third film based on Marvel's Guardians of the Galaxy. The third film based on Marvel's
                   Guardians of the Galaxy"
             />
         </div>
    );
};
