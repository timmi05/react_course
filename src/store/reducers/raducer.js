import {FETCH_MOVIES_BEGIN, FETCH_MOVIES_FAILURE, FETCH_MOVIES_SUCCESS, MOVIE_SELECTED} from '../actions/movies';

const initialState = {
    data: [],
    loading: false,
    error: null,
    isProductSelected: false,
    selectedProduct: {},
    value: '',
    searchBy: 'title',
    sortBy: 'vote_average'
};
export const reducer = (state = initialState, action) => {
    switch (action.type) {
        case FETCH_MOVIES_BEGIN:
            return {
                ...state,
                loading: true,
                error: null
            };

        case FETCH_MOVIES_SUCCESS:
            return {
                ...state,
                loading: false,
                data: action.payload.movies
            };

        case FETCH_MOVIES_FAILURE:
            return {
                ...state,
                loading: false,
                error: action.payload.error,
                data: []
            };

        case MOVIE_SELECTED:
            return {
                ...state,
                isProductSelected: true,
                selectedProduct: action.payload
            };
        case 'MOVE_TO_SEARCH':
            return {
                ...state,
                isProductSelected: false,
                selectedProduct: {}
            };
        case 'SEARCH_MOVIES':
            return {
                ...state,
                value: action.payload.value,
                searchBy: action.payload.searchBy
            };
        case 'SORT_BY':
            return {
                ...state,
                sortBy: action.payload
            };
        default:
            return state;
    }
};
