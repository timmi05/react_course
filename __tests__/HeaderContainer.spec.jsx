import React from 'react';
import {shallow} from 'enzyme'
import HeaderContainer from "../src/HeaderContainer";
import {createStore} from 'redux';
import {reducer} from '../src/store/reducers/raducer.js';

describe('HeaderContainer rendering', () => {
    const store = createStore(reducer);

    it('renders if product is not selected', () => {
        const component = shallow(<HeaderContainer
            store={store}
        />);
        expect(component).toMatchSnapshot();
    });
});
