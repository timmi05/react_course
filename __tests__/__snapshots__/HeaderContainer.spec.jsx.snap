// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer rendering renders if product is not selected 1`] = `
<ContextProvider
  value={null}
>
  <HeaderContainer
    isProductSelected={false}
    onMoveToSearch={[Function]}
    selectedProduct={Object {}}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</ContextProvider>
`;
